Source/WebCore/dom/ContentVisibilityDocumentState.cpp

```diff
@@ -230,4 +232,20 @@ void ContentVisibilityDocumentState::removeViewportProximity(const Element& elem
     m_elementViewportProximities.remove(element);
 }
 
+void ContentVisibilityDocumentState::elementContentVisibilityDidChange(const Element& element, const ContentVisibility& oldValue, const ContentVisibility& newValue)
+{
+    ASSERT(oldValue != newValue);
+    if (oldValue == newValue || newValue != ContentVisibility::Visible)
+        return;
+    auto animations = element.document().matchingAnimations([&] (Element& target) -> bool {
+        return element.contains(target);
+    });
+    for (auto animation : animations) {
+        if (auto* declarativeAnimation = dynamicDowncast<DeclarativeAnimation>(animation.get())) {
+            if (auto* timeline = declarativeAnimation->timeline())
+                timeline->animationTimingDidChange(*declarativeAnimation);
+        }
+    }
```

Comment:
Looking at the code again, this is not super optimal since we now have two loops. Calling `matchingAnimations` forces a layout which is something important when `Document.getAnimations()` is called via the bindings but not in this case, and in fact could be harmful.

I apologize for suggesting this, upon closer inspection I think your previous code is better. Ideally, we'd find a cleaner way to iterate over animations that both `Document::matchingAnimations()` and this new method would both use, but whatever you had before is satisfactory to me.

Reply:
No worries, I reverted the change.